### DISCUSSION TITLE
Rect of CentralPanel

### DISCUSSION THREAD

**Discussion by hendrix63** [emilk/egui#4765] [Q&A] [answered]
It's probably a newbie question but I want to create a setup as shown in the image. My goal is to split the CentralPanel vertically to show 2 tables each taking up 50% of the space regardless of the number of rows in the table.

 To do this I want to get the rect of the CentralPanel and use manual layout to position the tables. How do I get the rect of the CentralPanel?
<img width="389" alt="egui" src="https://github.com/emilk/egui/assets/57320220/1f1acc8b-eae1-42b3-a7ef-9a3db78b5fbf">

**Comment by YgorSouza** [comment, marked as answer]
[Ui::max_rect](https://docs.rs/egui/0.28.0/egui/struct.Ui.html#method.max_rect) will give you the current UI's rect. When you add a widget that fills all the available space, that is the size it will take. Also, you can use [Ui::columns](https://docs.rs/egui/0.28.0/egui/struct.Ui.html#method.columns) if you want to split the UI evenly. For uneven splits, there is also [egui_extras::Strip](https://docs.rs/egui_extras/latest/egui_extras/struct.StripBuilder.html).

```rs
    fn update(&mut self, ctx: &egui::Context, _frame: &mut eframe::Frame) {
        egui::TopBottomPanel::top("top").show(ctx, |ui| {
            ui.label("Top");
        });
        egui::SidePanel::left("left").show(ctx, |ui| {
            ui.label("Side");
        });
        egui::CentralPanel::default().show(ctx, |ui| {
            let rect = ui.max_rect();
            ui.painter()
                .debug_rect(rect, egui::Color32::BLUE, "Central panel");
            ui.columns(2, |cols| {
                for (i, ui) in cols.iter_mut().enumerate() {
                    let rect = ui.max_rect();
                    ui.painter()
                        .debug_rect(rect, egui::Color32::RED, format!("Column {i}"));
                }
            });
        });
    }
```

![image](https://github.com/emilk/egui/assets/43298013/47270d92-49e0-48a0-aecf-9c1f3dbbb6c8)

The space between the columns is controlled by the `item_spacing` field. If you want to remove it:

```rs
ui.spacing_mut().item_spacing = egui::Vec2::ZERO;
```

Then you might want to add it back inside each column so the widgets are spaced nicely.

**Reply by hendrix63** [reply]
Thank you so much. I do not want to use columns as the tables should be on top of each other but your answer got me in the right direction. This will help me get it done.
